game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -167,7 +169,7 @@
 
 /** Main frame for the triple a game. */
 @Log
-public final class TripleAFrame extends JFrame {
+public final class TripleAFrame extends JFrame implements KeyBindings {
```

Comment:
I don't quite agree with the term "KeyBindings" here.
In my opinion the interfaces name should make clear that it only supplies a collection of bindings, but doesn't actually manage them.
Maybe that's just me being silly, but just wanted to point out that this confused me for a moment

Reply:
The thinking was an implementor "HasKeyBeindings". Any suggestions for alternatives?

Reply:
`KeyControllable`?

Reply:
Not of a fan of that one,  `KeyBindingListener` maybe. TBH I'm pretty happy with `KeyBindings`

Reply:
`KeyBindingListener` it is! 😄

Reply:
Are we sure that is a better name?
A listener responds to an event, in this case the event could be interpreted as the key binding itself. IE: when a key is bound, fire this listener, rather than responding to a key press event. In that case we really want a 'KeyPressListener'; and we also have the issue that we have an API returning many listeners.

`KeyPressHandler` is maybe an okay name. Or maybe `KeyBindingSupplier`, but I feel like we're getting into the kingdom of nouns problem.

Reply:
`KeyBindingSupplier` is pretty descriptive, because well it implements supplier.
I'd go with that one

Reply:
There is a small issue of singular vs plural as this interface is returning many key bindings and not just one. It reads well enough though, I'm okay going with that name.